marathon-interface/src/main/scala/mesosphere/marathon/interface/auth/Authenticator.scala

```diff
@@ -0,0 +1,39 @@
+package mesosphere.marathon.interface.auth
+
+import mesosphere.marathon.interface.http.{ HttpResponse, HttpRequest }
+import mesosphere.marathon.interface.plugin.Plugin
+
+import scala.concurrent.Future
+
+/**
+  * Base trait for all authenticator implementations.
+  * Every authenticator can get data from an HTTP request and derive an Identity.
+  * The Identity usually is maintained in a separate Identity and Access Management System (e.g. LDAP)
+  *
+  * The Identity is passed to the [[Authorizer]].
+  * It should hold all information to make decisions possible for authorization.
+  *
+  * The Resources will make sure, that the [[Authenticator#authenticate]]
+  * will call this method only once per request.
+  *
+  * The authenticate method method is most probably one that will query external identity providers,
+  * and uses a Future as result.
+  */
+trait Authenticator extends Plugin {
```

Comment:
I don't think the Plugin trait is technically necessary. For me it is a Marker that says: this is a Plugin interface. We do provide a lot of traits here and you don't see provided/required.